options.go

```diff
@@ -0,0 +1,267 @@
+package envbuilder
+
+import (
+	"github.com/coder/serpent"
+)
+
+// Options contains the configuration for the envbuilder.
+type Options struct {
+	SetupScript          string
+	InitScript           string
+	InitCommand          string
+	InitArgs             string
+	CacheRepo            string
+	BaseImageCacheDir    string
+	LayerCacheDir        string
+	DevcontainerDir      string
+	DevcontainerJSONPath string
+	DockerfilePath       string
+	BuildContextPath     string
+	CacheTTLDays         int64
+	DockerConfigBase64   string
+	FallbackImage        string
+	ExitOnBuildFailure   bool
+	ForceSafe            bool
+	Insecure             bool
+	IgnorePaths          []string
+	SkipRebuild          bool
+	GitURL               string
+	GitCloneDepth        int64
+	GitCloneSingleBranch bool
+	GitUsername          string
+	GitPassword          string
+	GitHTTPProxyURL      string
+	WorkspaceFolder      string
+	SSLCertBase64        string
+	ExportEnvFile        string
+	PostStartScriptPath  string
+}
+
+// Generate CLI options for the envbuilder command.
+func (o *Options) CLI() serpent.OptionSet {
+	return serpent.OptionSet{
+		{
+			Flag:  "setup-script",
+			Env:   "SETUP_SCRIPT",
+			Value: serpent.StringOf(&o.SetupScript),
+			Description: "SetupScript is the script to run before the init script. It runs as " +
+				"the root user regardless of the user specified in the devcontainer.json " +
+				"file.\n\nSetupScript is ran as the root user prior to the init script. " +
+				"It is used to configure envbuilder dynamically during the runtime. e.g. " +
+				"specifying whether to start systemd or tiny init for PID 1.",
+		},
+		{
+			Flag:        "init-script",
+			Env:         "INIT_SCRIPT",
+			Default:     "sleep infinity",
+			Value:       serpent.StringOf(&o.InitScript),
+			Description: "InitScript is the script to run to initialize the workspace.",
```

Comment:
Stylistic nit: prefixing the descriptions with the variable name is unnecessary, and likely a consequence of moving the text from comments (which follow Go's conventions) to here.

`InitScript` isn't a valid CLI option; `init-script` and `INIT_SCRIPT` are.

Example refactor:

```suggestion
			Description: "The script to run to initialize the workspace",
```

Reply:
For context, this is what it looks like in the `--help` output:

```
       string, $INIT_SCRIPT (default: sleep infinity)
          InitScript is the script to run to initialize the workspace.
```

As a follow-up, it might make sense to add a test for this against a `.golden` file.

Reply:
I'd personally like to see golden file generation/validation as a part of serpents design (subpackage?), would be a cool feature for applications that need strict guarantees.

Reply:
@BrunoQuaresma There's usually just 2 components to golden file testing:

1. Capture output from the command(s) you want to verify
2. Write update, or verify output against existing file (e.g. `testdata/help/envbuilder--help.golden`)

To achieve 1) you set the serpent invocation stdout/stderr output to a buffer (e.g. `bytes.Buffer`) and run it.

To achieve 2) you add `var updateGoldenFiles = flag.Bool("update", false, "update .golden files")` to the test file:
- If true, write the output from 1) to the golden file
- If false, read output from the golden file and do `if diff := cmp.Diff(want, got); diff != "" { require.Failf(t, "mismatch (-want +got):\n%s", diff) }`

There's some precedent in `coder/coder` but I wouldn't recommend copying it directly as there's a lot we don't need from there: https://github.com/coder/coder/blob/main/cli/clitest/golden.go